transport/grpc/outbound.go

```diff
@@ -131,6 +134,21 @@ func (o *Outbound) Call(ctx context.Context, request *transport.Request) (*trans
 	}, invokeErr
 }
 
+func validateRequest(req *transport.Request) error {
+	for _, v := range req.Headers.Items() {
+		// from https://httpwg.org/specs/rfc7540.html#rfc.section.10.3:
+		// HTTP/2 allows header field values that are not valid.
+		// While most of the values that can be encoded will not alter header field parsing,
+		// carriage return (CR, ASCII 0xd), line feed (LF, ASCII 0xa),
+		// and the zero character (NUL, ASCII 0x0) might be exploited
+		// by an attacker if they are translated verbatim.
+		if strings.ContainsRune(v, '\r') || strings.ContainsRune(v, '\n') || strings.ContainsRune(v, '\x00') {
```

Comment:
It may be faster to do a `strings.ContainsAny`, https://golang.org/pkg/strings/#ContainsAny

That way we only scan the string once (vs 3 times atm)

Reply:
yup this is much better!